src/register/RegistrationPage.jsx

```diff
@@ -251,221 +198,33 @@ const RegistrationPage = (props) => {
-        setEmailSuggestion({ suggestion: '', type: '' });
-        break;
-      case 'username':
-        setFormFields(prevState => ({ ...prevState, username: suggestion }));
-        props.resetUsernameSuggestions();
-        break;
-      default:
-        break;
-    }
-  };
-
-  const handleEmailSuggestionClosed = () => setEmailSuggestion({ suggestion: '', type: '' });
-
-  const handleUsernameSuggestionClosed = () => props.resetUsernameSuggestions();
-
   const handleOnChange = (event) => {
     const { name } = event.target;
-    let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+    const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
     if (registrationError[name]) {
-      clearBackendError(name);
+      dispatch(clearRegistrationBackendError(name));
       setErrors(prevErrors => ({ ...prevErrors, [name]: '' }));
     }
-    if (name === 'username') {
-      if (value.length > 30) {
-        return;
-      }
-      if (value.startsWith(' ')) {
-        value = value.trim();
-      }
-    }
-
     setFormFields(prevState => ({ ...prevState, [name]: value }));
   };
 
-  const handleOnBlur = (event) => {
-    const { name, value } = event.target;
+  const handleErrorChange = (fieldName, error) => {
     if (registrationEmbedded) {
-      if (name === 'name') {
-        validateInput(
-          name,
-          value,
-          { name: formFields.name, username: formFields.username, form_field_key: name },
-          !validationApiRateLimited,
-          false,
-        );
-      }
-      return;
-    }
-    const payload = {
-      name: formFields.name,
-      email: formFields.email,
-      username: formFields.username,
-      password: formFields.password,
-      form_field_key: name,
-    };
-
-    setFocusedField(null);
-    validateInput(name, name === 'password' ? formFields.password : value, payload, !validationApiRateLimited);
-  };
-
-  const handleOnFocus = (event) => {
-    const { name, value } = event.target;
-    setErrors(prevErrors => ({ ...prevErrors, [name]: '' }));
-    clearBackendError(name);
-    // Since we are removing the form errors from the focused field, we will
-    // need to rerun the validation for focused field on form submission.
-    setFocusedField(name);
-
-    if (name === 'username') {
-      props.resetUsernameSuggestions();
-      // If we added a space character to username field to display the suggestion
-      // remove it before user enters the input. This is to ensure user doesn't
-      // have a space prefixed to the username.
-      if (value === ' ') {
-        setFormFields(prevState => ({ ...prevState, [name]: '' }));
+      setTemporaryErrors(prevErrors => ({
+        ...prevErrors,
+        [fieldName]: error,
+      }));
+      if (error === '' && errors[fieldName] !== '') {
+        setErrors(prevErrors => ({
+          ...prevErrors,
+          [fieldName]: error,
+        }));
```

Comment:
I thought we were maintaining `temporaryErrors` state for the embedded registration form. Why do we need to set the `Errors` state?

Reply:
This is for the scenario in embedded experience when the user presses the submit button and all of the overs are rendered. Now if the user focuses on any field the error should be cleared. To handle this scenario, I added this code.